From 538be7bce0bc2eb5d8ab1a1a222b69add761648f Mon Sep 17 00:00:00 2001
From: Robert Hodaszi <robert.hodaszi@digi.com>
Date: Wed, 19 Jan 2022 17:30:33 +0100
Subject: [PATCH] Fix build error because of conflicting env variable name

ROOTDIR is used by ucfront. If hostap overrides that, ucfront's libdir
output becomes invalid.
---
 src/build.rules | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/build.rules b/src/build.rules
index acda884..281c870 100644
--- a/src/build.rules
+++ b/src/build.rules
@@ -5,11 +5,11 @@ all: _all
 .SUFFIXES:
 
 # setup some variables
-ROOTDIR := $(dir $(lastword $(MAKEFILE_LIST)))
-ROOTDIR := $(dir $(ROOTDIR:%../src/=%))../
-BUILDDIR ?= $(abspath $(ROOTDIR)build)
+HOSTAP_ROOTDIR := $(dir $(lastword $(MAKEFILE_LIST)))
+HOSTAP_ROOTDIR := $(dir $(HOSTAP_ROOTDIR:%../src/=%))../
+BUILDDIR ?= $(abspath $(HOSTAP_ROOTDIR)build)
 BUILDDIR := $(BUILDDIR:%/=%)
-ABSROOT := $(abspath $(ROOTDIR))
+ABSROOT := $(abspath $(HOSTAP_ROOTDIR))
 ifeq ($(origin OUT),command line)
 _PROJ := $(OUT:%/=%)
 _PROJ := $(_PROJ:$(BUILDDIR)/%=%)
@@ -82,14 +82,14 @@ _DIRS := $(BUILDDIR)/$(PROJ)
 _make_dirs:
 	@mkdir -p $(_DIRS)
 
-$(BUILDDIR)/$(PROJ)/src/%.o: $(ROOTDIR)src/%.c $(CONFIG_FILE) | _make_dirs
+$(BUILDDIR)/$(PROJ)/src/%.o: $(HOSTAP_ROOTDIR)src/%.c $(CONFIG_FILE) | _make_dirs
 	$(Q)$(CC) -c -o $@ $(CFLAGS) $<
 	@$(E) "  CC " $<
 $(BUILDDIR)/$(PROJ)/%.o: %.c $(CONFIG_FILE) | _make_dirs
 	$(Q)$(CC) -c -o $@ $(CFLAGS) $<
 	@$(E) "  CC " $<
 # for the fuzzing tests
-$(BUILDDIR)/$(PROJ)/wpa_supplicant/%.o: $(ROOTDIR)wpa_supplicant/%.c $(CONFIG_FILE) | _make_dirs
+$(BUILDDIR)/$(PROJ)/wpa_supplicant/%.o: $(HOSTAP_ROOTDIR)wpa_supplicant/%.c $(CONFIG_FILE) | _make_dirs
 	$(Q)$(CC) -c -o $@ $(CFLAGS) $<
 	@$(E) "  CC " $<
 
@@ -100,9 +100,9 @@ lib_phony:
 # nothing
 
 $(BUILDDIR)/$(PROJ)/%.a: $(CONFIG_FILE) lib_phony
-	$(Q)$(MAKE) -C $(ROOTDIR)$(dir $(@:$(BUILDDIR)/$(PROJ)/%=%)) OUT=$(abspath $(dir $@))/
+	$(Q)$(MAKE) -C $(HOSTAP_ROOTDIR)$(dir $(@:$(BUILDDIR)/$(PROJ)/%=%)) OUT=$(abspath $(dir $@))/
 
-BUILDOBJ = $(patsubst %,$(BUILDDIR)/$(PROJ)/%,$(patsubst $(ROOTDIR)%,%,$(1)))
+BUILDOBJ = $(patsubst %,$(BUILDDIR)/$(PROJ)/%,$(patsubst $(HOSTAP_ROOTDIR)%,%,$(1)))
 
 .PHONY: common-clean
 common-clean:
-- 
2.27.0

